DOCS-#3863: Add usage guide to documentation (#3864)
modin-project/modin: docs/usage_guide/advanced_usage/index.rst

   :titlesonly:
   :hidden:

   /flow/modin/distributed/dataframe/pandas
   spreadsheets_api
   progress_bar
   modin_sql
   modin_xgboost
   modin_in_the_cloud

.. meta::
    :description lang=en:
        Description of Modin's advanced features.

Modin aims to not only optimize pandas, but also provide a comprehensive,
integrated toolkit for data scientists. We are actively developing data science tools

> such as DataFrame - spreadsheet integration, DataFrame algebra, progress bars, SQL queries

Reviewer: ```suggestion
such as DataFrame spreadsheet integration, DataFrame algebra, progress bars, SQL queries
```

Reviewer: add links to appropriate repos/docs

Author: All the links are in corresponding sections below. Does that make sense?

> on DataFrames, and more. Join the `Discourse`_ for the latest updates!

Reviewer: ```suggestion
on DataFrames, and more. Join us on `Slack`_ and `Discourse`_ for the latest updates!
```
Note add link to Slack: https://modin.org/slack.html

Author: Added the link.

- :py:func:`~modin.experimental.pandas.read_pickle_distributed`  -- read multiple files in a directory
- :py:meth:`~modin.experimental.pandas.DataFrame.to_pickle_distributed` -- write to multiple files in a directory

DataFrame partitioning API
--------------------------

Modin DataFrame provides an API to directly access partitions: you can extract physical partitions from
a :py:class:`~modin.pandas.dataframe.DataFrame`, modify their structure by reshuffling or applying some
functions, and create a DataFrame from those modified partitions. Visit
:doc:`pandas partitioning API </flow/modin/distributed/dataframe/pandas>` documentation to learn more.

> Modin Spreadsheet API: Render Dataframes as Spreadsheets

Reviewer: ```suggestion
Modin Spreadsheet API
```

--------------------------------------------------------

The Spreadsheet API for Modin allows you to render the dataframe as a spreadsheet to easily explore
your data and perform operations on a graphical user interface. The API also includes features for recording
the changes made to the dataframe and exporting them as reproducible code. Built on top of Modin and SlickGrid,
the spreadsheet interface is able to provide interactive response times even at a scale of billions of rows.
See our `Modin Spreadsheet API documentation`_ for more details.

.. figure:: /img/modin_spreadsheet_mini_demo.gif
   :align: center
   :width: 650px
   :height: 350px

Progress Bar
------------

Visual progress bar for Dataframe operations such as groupby and fillna, as well as for file reading operations such as

> read_csv. Built using the tqdm library and Ray execution engine. See `Progress Bar documentation`_ for more details.

Reviewer: link to tqdm: https://github.com/tqdm/tqdm

Author: added

.. figure:: /img/progress_bar_example.png
   :align: center

Dataframe Algebra
-----------------

A minimal set of operators that can be composed to express any dataframe query for use in query planning and optimization.
See our `paper`_ for more information, and full documentation is coming soon!

SQL on Modin Dataframes
------------------------

> .. figure:: /img/modin_sql_example.png
>    :align: center
> 
> Read about Modin Dataframe support for SQL queries in this recent `blog post`_. Check out the `Modin SQL documentation`_ as well!

Reviewer: Moved to text first then image

```suggestion
Read about Modin Dataframe support for SQL queries in this recent `blog post`_. Check out the `Modin SQL documentation`_ as well!

.. figure:: /img/modin_sql_example.png
   :align: center
```
```

Distributed XGBoost on Modin
----------------------------

Modin provides an implementation of distributed XGBoost machine learning algorithm on Modin DataFrames. See our
:doc:`Distributed XGBoost on Modin documentation <modin_xgboost>` for details about installation and usage, as well as

> :doc:`Modin XGBoost architecture documentation </flow/modin/experimental/xgboost>` for information about implementation and

Reviewer: link to this blogpost? 
https://medium.com/intel-analytics-software/distributed-xgboost-with-modin-on-ray-fc17edef7720

Author: added

- :py:func:`~modin.experimental.pandas.read_pickle_distributed`  -- read multiple files in a directory
- :py:meth:`~modin.experimental.pandas.DataFrame.to_pickle_distributed` -- write to multiple files in a directory

DataFrame partitioning API
--------------------------

> Modin DataFrame provides an API to manipulate with its partitions: you can extract raw partitions from
> :py:class:`~modin.pandas.dataframe.DataFrame`, modify their structure by reshuffling or applying some
> functions, and then inject them back to the DataFrame. Visit
> :doc:`partitioning API documentation </flow/modin/distributed/dataframe/pandas>` to learn more.

Reviewer 2: ```suggestion
Modin DataFrame provides an API to directly access partitions: you can extract physical partitions from
a :py:class:`~modin.pandas.dataframe.DataFrame`, modify their structure by reshuffling or applying some
functions, and create a DataFrame from those modified partitions. Visit
:doc:`partitioning API documentation </flow/modin/distributed/dataframe/pandas>` to learn more.
```
